src/components/dialog.js

```diff
@@ -230,12 +234,33 @@ export class Dialog {
   }
 
   /**
-   * Whether to display loading indicator.
-   * @param {boolean} isLoading
+   * Transitions to the next view
+   * @private
    */
-  setLoading(isLoading) {
-    if (isLoading) {
+  entryTransitionToNextView_() {
+    if (this.view_ && this.view_.hasLoadingIndicator()) {
+      // Temporarily cache the old view
+      this.previousProgressView_ = this.view_;
+    } else {
+      // Since loading indicator will be shown, remove contents of old view
+      removeChildren(this.getContainer());
+      // When loading indicator was not displayed in the previous view,
+      // loading indicator must be displayed while transitioning to new view.      
       this.loadingView_.show();
+    }
+  }
+
+   /**
+   * Transition out of an old view
+   * @private
+   */
+  exitTransitionFromOldView_() {
+    //When a loading indicator is present, hide it.
+    this.loadingView_.hide();
```

Comment:
Loading indicator is hidden twice: let's leave only one. FWIW, always trying to hide both: loading indicator and previous view seemed like an easier to read option to me.